From 6c988491aeb33c3e947b36fce6d3e74ee885f382 Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 22:43:38 -0400
Subject: [infiniband] cxgb4: Fix skb_leak in reject_cr()

Message-id: <1410475447-94392-80-git-send-email-svemuri@redhat.com>
Patchwork-id: 93802
O-Subject: [RHEL7.1 PATCH BZ 1124947 079/108] RDMA/cxgb4: Fix skb_leak in reject_cr()
Bugzilla: 1124947
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Based on origninal work by Steve Wise <swise@opengridcomputing.com>

Signed-off-by: Steve Wise <swise@opengridcomputing.com>
Signed-off-by: Hariprasad Shenai <hariprasad@chelsio.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 233b43010330ed8cf39cf636880017df3e33f102)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/cxgb4/cm.c b/drivers/infiniband/hw/cxgb4/cm.c
index 60646ab..4e59fd0 100644
--- a/drivers/infiniband/hw/cxgb4/cm.c
+++ b/drivers/infiniband/hw/cxgb4/cm.c
@@ -2180,7 +2180,6 @@ static void reject_cr(struct c4iw_dev *dev, u32 hwtid, struct sk_buff *skb)
  PDBG("%s c4iw_dev %p tid %u\n", __func__, dev, hwtid);
  BUG_ON(skb_cloned(skb));
  skb_trim(skb, sizeof(struct cpl_tid_release));
- skb_get(skb);
  release_tid(&dev->rdev, hwtid, skb);
  return;
 }
-- 
1.7.1